File: datashield/smoke_auth_test.py
import json
import urllib.request

url = 'http://localhost:5000/api/auth/register'
data = json.dumps({'name': 'Test User', 'email': 'testuser@example.com', 'password': 'Password123'}).encode('utf-8')
req = urllib.request.Request(url, data=data, headers={'Content-Type': 'application/json'})
try:
    with urllib.request.urlopen(req, timeout=15) as r:
        print('STATUS', r.status)
        print(r.read().decode())
except Exception as e:
    print('ERROR', e)

File: datashield/smoke_auth_test.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const data = JSON.parse(fs.readFileSync(path.join(__dirname, 'auth_register.json'), 'utf8'));

const url = 'http://localhost:5000/api/auth/register';

try {
  const res = await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data),
  });
  const text = await res.text();
  console.log('STATUS', res.status);
  console.log(text);
} catch (err) {
  console.error('ERROR', err);
}
